### PR TITLE
add pyfuse3 as an alternative lowlevel fuse implementation

### PR DESCRIPTION
**Pull request by ThomasWaldmann**

### CODE REVIEW

**Review by Gu1nness — CHANGES_REQUESTED** [review]
* I would add a test where BORG_FUSE_IMPL is set to an empty variable to ensure that it crashed the way we want to avoid regressions in the future.
 * The documentation needs to be updated to be indented the way we should.
 * I have run all the tests and it seems to be working (fuse2 and fuse3)

File: src/borg/fuse_impl.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        except ImportError:
+            pass
+        else:
+            has_llfuse = True
+            has_pyfuse3 = False
+            break
+    elif FUSE_IMPL == 'none':
+        pass
+    else:
+        raise RuntimeError("unknown fuse implementation in BORG_FUSE_IMPL: '%s'" % BORG_FUSE_IMPL)
```
[the remark this answers is not in the document]
Comment (ThomasWaldmann) [reply src/borg/fuse_impl.py:32]: I can try raising `Error`, that should give a simple error msg.

File: docs/usage/general/environment.rst.inc [hunk 2]
```diff
@@ -64,6 +64,16 @@ General:
...
+    BORG_FUSE_IMPL
+        Choose the lowlevel FUSE implementation borg shall use for ``borg mount``.
+	This is a comma-separated list of implementation names, they are tried in the
+	given order, e.g.:
+
+        - ``pyfuse3,llfuse``: default, first try to load pyfuse3, then try to load llfuse.
+        - ``llfuse,pyfuse3``: first try to load llfuse, then try to load pyfuse3.
+	- ``pyfuse3``: only try to load pyfuse3
+	- ``llfuse``: only try to load llfuse
+	- ``none``: do not try to load an implementation
```
[the remark this answers is not in the document]
Comment (ThomasWaldmann) [reply docs/usage/general/environment.rst.inc:76]: will check...

[the remark this answers is not in the document]
Comment (ThomasWaldmann) [reply docs/usage/general/environment.rst.inc:76]: confirmed / fixed.

File: src/borg/fuse_impl.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ThomasWaldmann) [reply src/borg/fuse_impl.py:32]: nope, that doesn't help (isn't catched by the usual exception handler).

but i don't think this is a big problem anyway, not many people will have a need to set this variable, because the default will work. it is mostly for unit testing, so we can test a specific implementation.

**Comment by ThomasWaldmann** [comment]
Squashed the fixup, reword another commit comment. Will merge after tests complete.
